app/models/comment.rb

```diff
@@ -148,7 +152,7 @@ def evaluate_markdown
   end
 
   def adjust_comment_parent_based_on_depth
-    self.parent_id = parent.descendant_ids.last if parent && (parent.depth > 1 && parent.has_children?)
+    self.parent_id = parent.descendant_ids.last if root_exists? && (parent.depth > 1 && parent.has_children?)
```

Comment:
There's something that I don't understand. `.parent` is the parent comment, which could be the root or not depending if the comment is the child of the root or a "grandchild". Doesn't this change the behavior of the method?

Reply:
I'm als not sure what happens to children comments if a parent comment gets deleted so i feel i don't have the necessary knowledge to review this correctly eheheh

Reply:
I thought parent and root were interchangeable and it was the comment that the current one branches off of. After doing some [research it looks like this is not the case](https://github.com/stefankroes/ancestry/blob/741042f40bd495cf196dd7ffb8e2afa5569e74fb/README.md#navigating-your-tree)
```
irb(main):002:0> c.root
Traceback (most recent call last):
        1: from (irb):2
ActiveRecord::RecordNotFound (Couldn't find Comment with 'id'=270539)
irb(main):003:0> c.parent
Traceback (most recent call last):
        2: from (irb):2
        1: from (irb):3:in `rescue in irb_binding'
ActiveRecord::RecordNotFound (Couldn't find Comment with 'id'=270539)
```

Ugh, I guess we have to safely check for parents as well. Good catch!

Reply:
> so i feel i don't have the necessary knowledge to review this correctly 

LOL that's what asking questions is for, so the author can put in the research they need to 😉